src/plugins/core/borders.ts

```diff
@@ -329,7 +330,7 @@ export class BordersPlugin extends CorePlugin<BordersPluginState> implements Bor
         destructive: false,
       });
     }
-    this.getRowsRange(sheetId)
+    range(0, this.getters.getNumberRows(sheetId) + totalDeletedRows + 1)
```

Comment:
It seems a bit strange that we need something different here than for columns removing, while it should probably be done in the same way :/
We can maybe use something like the `this.getColumnsWithBorders(sheetId)` method used in the `shiftBordersHorizontally`, to avoid adding a parameter in this method, with something like
```js
  private getRowsWithBorders(sheetId: UID): HeaderIndex[] {
    const sheetBorders = this.borders[sheetId]?.filter(isDefined);
    if (!sheetBorders) return [];
    return Object.keys(sheetBorders[0]).map((index) => parseInt(index, 10));
  }
```
(juste make sure taking the first item of `sheetBorders` is enough)
Any idea about this suggestion @rrahir ?

Reply:
@anhe-odoo Agreed, there doesn't seem to be a good reason not to use the same approach for both the columns and rows. The suggested `getRowsWithBorders` seems good :)